pkg/objstore/prefixed_bucket.go

```diff
@@ -0,0 +1,88 @@
+package objstore
+
+import (
+	"context"
+	"io"
+	"strings"
+)
+
+type PrefixedBucket struct {
+	bkt    Bucket
+	prefix string
+}
+
+func NewPrefixedBucket(bkt Bucket, prefix string) Bucket {
+	pbkt := &PrefixedBucket{bkt: bkt, prefix: strings.Trim(prefix, DirDelim)}
```

Comment:
The usage of `DirDelim` here is something that tricked me.
This is because we're stating that directories are going to be separated by `/`. I believe this would make filesystem bucket type to not work on Windows OS. But then I saw that metrics bucket [uses slashes](https://github.com/jademcosta/thanos/blob/787a97aafa2fe1dca7ae1e18dc75588991cc2f9d/pkg/objstore/objstore.go#L273), and as nobody complained, I believe Windows users doesn't run Thanos very often.
What is your opinion on that?

Reply:
Yeah, I think your observation is correct however I think we could still fix this. Could you open up an issue for this? I think we could mark it as a "good first issue" because it should be relatively straightforward? :thinking:

Reply:
Done: https://github.com/thanos-io/thanos/issues/5380